src/main/java/overlap_use_case/TimetableModelUnpacker.java

```diff
@@ -0,0 +1,95 @@
+package overlap_use_case;
+
+import entities.*;
+import retrieve_timetable_use_case.*;
+
+import java.util.ArrayList;
+
+/** A class responsible for converting between Timetable Model classes (that is, including its internal
+ * model classes) back into their entity forms. "Unpacks the model class" into the entity.
+ */
+public class TimetableModelUnpacker {
+
+    public static Session unpackSessionModel(SessionModel sessionModel){
+
+        Session session = new Session(sessionModel.getSessionType());
+
+        for (CourseModel courseModel : sessionModel.getCourses().values()){
+            session.addCourse(unpackCourseModelAsCalendarCourse(courseModel));
+        }
+
+        return session;
+    }
+
+    public static Timetable unpackTimeTable(TimetableModel timetable){
```

Comment:
Just a tiny little correction so that it falls in line with the name of the Timetable class: I would call it unpackTimetable.

Reply:
AUUGH (done!)